test/tests.spec.ts

```diff
@@ -588,6 +588,21 @@ describe("node-saml /", function () {
         expect(metadata).to.contain('AuthnRequestsSigned="true"');
       });
 
+      it("generateServiceProviderMetadata contains AuthnRequestsSigned=false", function () {
+        const samlConfig: SamlConfig = {
+          cert: TEST_CERT,
+          issuer: "http://example.serviceprovider.com",
+          callbackUrl: "http://example.serviceprovider.com/saml/callback",
+          identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+          authnRequestsSigned: false,
```

Comment:
When I pull this code and change `authnRequestsSigned: false` to `authnRequestsSigned: true`, the test still passes, so I must be missing something about the way you intend this to work.

Reply:
Yes I explained it here: 
https://github.com/node-saml/node-saml/pull/143#discussion_r938196687

This options is used in a "logical" way

If we use the option as a boolean, two weird case can appear:

- if we put `authnRequestsSigned: false` while request is signed ? -> it will be overridden, because in the code, we force `authnRequestsSigned` to be true
- if we put `authnRequestsSigned: true` but the request is not signed (for example a missing cert) -> the SAML metadata will not be correct, because the request isn't signed

To solve that, the option is: "if defined, put the options" (the explanation is in the `README.md`)

Reply:
I see what you're trying to do now. According to the [spec](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf) line 689, this should default to false. So, why not just include it all the time and let it get overridden when it should be true as it currently does? Put another way, what value is added by having the ability to _not_ include `AuthnRequestsSigned="false"`?

Reply:
Note that the change is for `AuthnRequestsSigned` line 740 in [spec](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf)

> why not just include it all the time and let it get overridden when it should be true as it currently does?

- If, for example, you want a smal metadata xml file, you don't want all an optional field
- If we include it all the time, I think we need to rewrite tests :/

You choose, I can implement the `include by default` if you want

Reply:
Yea, I like the idea that it is included by default. It doesn't add that much to the metadata and it makes it very clear what is going on. I've pushed up making this the default, but it needs some cleanup to remove the option, readme changes, etc. Would you please follow?

Reply:
done !